Unhandled exception caught in core executor java.lang.NoClassDefFoundError: com/codahale/metrics/JmxReporter

Hi,

After the datastax driver upgrade, I wanted to try out the heroic-1.0.2 with C*, but I got a newer error:
```
2020-02-12 10:02:55,379 ERROR c.s.h.HeroicCore [heroic-core-2] Unhandled exception caught in core executor java.lang.NoClassDefFoundError: com/codahale/metrics/JmxReporter
        at com.datastax.driver.core.Metrics.<init>(Metrics.java:146)
        at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:1667)
        at com.datastax.driver.core.Cluster.init(Cluster.java:214)
        at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:387)
        at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:366)
        at com.datastax.driver.core.Cluster.connect(Cluster.java:311)
        at com.spotify.heroic.metric.datastax.ManagedSetupConnection.lambda$construct$0(ManagedSetupConnection.java:97)
        at eu.toolchain.async.TinyAsync$2.run(TinyAsync.java:218)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.ClassNotFoundException: com.codahale.metrics.JmxReporter
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        ... 13 more

2020-02-12 10:02:55,379 ERROR c.s.h.HeroicCore [heroic-core-2] Unhandled exception caught in core executor java.lang.NoClassDefFoundError: com/codahale/metrics/JmxReporter
        at com.datastax.driver.core.Metrics.<init>(Metrics.java:146)
        at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:1667)
        at com.datastax.driver.core.Cluster.init(Cluster.java:214)
        at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:387)
        at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:366)
        at com.datastax.driver.core.Cluster.connect(Cluster.java:311)
        at com.spotify.heroic.metric.datastax.ManagedSetupConnection.lambda$construct$0(ManagedSetupConnection.java:97)
        at eu.toolchain.async.TinyAsync$2.run(TinyAsync.java:218)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.ClassNotFoundException: com.codahale.metrics.JmxReporter
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        ... 13 more

2020-02-12 10:02:55,382 ERROR c.s.h.HeroicCore [heroic-core-2] Exiting (code=2) 
2020-02-12 10:02:55,382 ERROR c.s.h.HeroicCore [heroic-core-2] Exiting (code=2)

--- a/Dockerfile
+++ b/Dockerfile
@@ -17,5 +17,6 @@ COPY --from=builder heroic-dist/build/libs/heroic-dist-0.0.1-SNAPSHOT-shaded.jar
 COPY example/heroic-memory-example.yml /heroic.yml
 COPY run-heroic.sh /usr/bin/heroic.sh
 
+ENV JVM_DEFAULT_ARGS -Dcom.datastax.driver.FORCE_NIO=true
 ENTRYPOINT ["/usr/bin/heroic.sh"]
 CMD ["/heroic.yml"]

--- a/metric/datastax/src/main/java/com/spotify/heroic/metric/datastax/ManagedSetupConnection.java
+++ b/metric/datastax/src/main/java/com/spotify/heroic/metric/datastax/ManagedSetupConnection.java
@@ -53,8 +53,10 @@ public class ManagedSetupConnection implements ManagedSetup<Connection> {
                                         "readTimeout", "consistencyLevel", "retryPolicy",
                                         "authentication", "poolingOptions" })
     public ManagedSetupConnection(final AsyncFramework async,
-                                  final Collection<InetSocketAddress> seeds, final Schema schema,
-                                  final boolean configure, final int fetchSize,
+                                  final Collection<InetSocketAddress> seeds,
+                                  final Schema schema,
+                                  final boolean configure,
+                                  final int fetchSize,
                                   final Duration readTimeout,
                                   final ConsistencyLevel consistencyLevel,
                                   final RetryPolicy retryPolicy,
@@ -88,8 +90,8 @@ public class ManagedSetupConnection implements ManagedSetup<Connection> {
                 .withRetryPolicy(retryPolicy)
                 .withQueryOptions(queryOptions)
                 .withSocketOptions(socketOptions)
-                .withLoadBalancingPolicy(new TokenAwarePolicy(new RoundRobinPolicy()));
-            // @formatter:on
+                .withLoadBalancingPolicy(new TokenAwarePolicy(new RoundRobinPolicy()))
+                .withoutJMXReporting();
 
             authentication.accept(cluster);
             poolingOptions.apply(cluster);
@@ -98,16 +100,11 @@ public class ManagedSetupConnection implements ManagedSetup<Connection> {
         });
 
         if (configure) {
-            session = session.lazyTransform(s -> {
-                return schema.configure(s).directTransform(i -> s);
-            });
+            session = session.lazyTransform(s -> schema.configure(s).directTransform(i -> s));
         }
 
-        return session.lazyTransform(s -> {
-            return schema.instance(s).directTransform(schema -> {
-                return new Connection(s, schema);
-            });
-        });
+        return session.lazyTransform(
+            s -> schema.instance(s).directTransform(schema -> new Connection(s, schema)));
     }
 
     @Override

